test/test_simulation_results.jl

```diff
@@ -690,10 +689,26 @@ function test_simulation_results_from_file(path::AbstractString, c_sys5_hy_ed, c
     @test get_system(results_uc) === nothing
     @test length(read_realized_variables(results_uc)) == length(UC_EXPECTED_VARS)
 
+    results_ed = get_decision_problem_results(results, "ED")
+    @test isnothing(get_system(results_ed))
+
+    results_ed = get_decision_problem_results(results, "ED"; populate_system = true)
+    @test !isnothing(get_system(results_ed))
+    @test PSY.get_units_base(get_system(results_ed)) == "NATURAL_UNITS"
+
     @test_throws IS.InvalidValue set_system!(results_uc, c_sys5_hy_ed)
     set_system!(results_ed, c_sys5_hy_ed)
     set_system!(results_uc, c_sys5_hy_uc)
 
+    results_ed = get_decision_problem_results(
+        results,
+        "ED";
+        populate_system = true,
+        populate_units = IS.UnitSystem.DEVICE_BASE,
+    )
+    @test !isnothing(PSI.get_system(results_ed))
+    @test PSY.get_units_base(get_system(results_ed)) == "DEVICE_BASE"
+
```

Comment:
Can you also test that the errors work as expected when we can't find the system file and when we set `populate_units` without setting `populate_system`?

Reply:
once this test is implemented we can merge

Reply:
Hi @GabrielKS, could you confirm or deny the implementation of the test?